### PR TITLE
Feature/create food model

### PR DESCRIPTION
**Pull request by gtekle**
## In this pull request, I have:
- created `Food` model
- added `Food` model `validation`
- added `Food` Model `spec`

### CODE REVIEW

File: db/migrate/20220516173040_create_foods.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateFoods < ActiveRecord::Migration[7.0]
+  def change
+    create_table :foods do |t|
+      t.string :name, null: false
+      t.string :measurement_unit, null: false
+      t.decimal :price, default: 0.0
+      t.references :user, null: false, foreign_key: true
```
Comment (guerreiropedr0) [inline db/migrate/20220516173040_create_foods.rb:7]: [OPTIONAL] This is more of a question, in the ERD it says user_id and not user, shouldn't we keep it to the diagram?

**Review by guerreiropedr0 — APPROVED** [review]
Hi @gtekle 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:


nodding-yes

## Optional suggestions
- Check the single comments :smiley: 

##### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

##### Cheers and Happy coding!:clap::clap::clap:
##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------

File: db/migrate/20220516173040_create_foods.rb (hunk 1 above)
Comment (gtekle) [reply db/migrate/20220516173040_create_foods.rb:7]: Yes, you are right it should be user_id. and Active Record creates user_id column as foreign key, if you give it a model name. This is a  snapshot of the foods table in Postgres:
user_id-column-in-foods-table

File: spec/models/food_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+require 'rails_helper'
+
+RSpec.describe Food, type: :model do
+  let(:user) { FactoryBot.create(:user, name: 'Tekle') }
+  let(:food) do
+    FactoryBot.create(:food,
+                      name: 'Apple',
+                      measurement_unit: 'grams',
+                      price: 1.15,
+                      user: user)
```
[the remark this answers is not in the document]
Comment (gtekle) [reply spec/models/food_spec.rb:30]: It assigns the Model id by default. But, it is also possible to explicitly assign it.

**Comment by gtekle** [comment]
Thanks :pray:  @guerreiropedr0

**Pull request merged**